### PR TITLE
Fix ES permissions; CI/CD/env infra improvements

### PR DESCRIPTION
**Pull request by kislyuk**
Sorry this is a hodge-podge of stuff. Working on un-screwing up our daemons and deployment infra.

- Update and parameterize Elasticsearch IAM access policies for Lambdas

- Begin centralized IAM policy templates and management scripts

- Begin centralized environment script

- Begin centralized Makefile util library

- Add check to Makefiles to ensure environment script is sourced before running commands

- Add docs for the deployment and CI/CD config process in the main readme

### CODE REVIEW

File: iam/policy-templates/ci-cd.json [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        "s3:GetReplicationConfiguration",
+        "s3:ListAllMyBuckets",
+        "s3:ListBucket",
+        "s3:ListBucketMultipartUploads",
+        "s3:ListBucketVersions",
+        "s3:ListMultipartUploadParts",
+        "s3:ListObjects"
+      ],
+      "Resource": [
+        "arn:aws:s3:::hca-dss-test-src",
```
Comment (ttung) [inline iam/policy-templates/ci-cd.json]: why is DSS_S3_TEST_BUCKET parameterized and not this?

Comment (sampierson) [inline iam/policy-templates/ci-cd.json]: Yes you can pick it up from the environment as $DSS_S3_TEST_SRC_DATA_BUCKET

Comment (kislyuk) [inline iam/policy-templates/ci-cd.json]: Fixed, thanks!

**Comment by kislyuk** [comment]
Merging this since I think I've addressed the comments and need this for further integration work.

**Pull request merged**